sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/package.scala

```diff
@@ -142,7 +142,10 @@ package object util extends Logging {
     "`" + name.replace("`", "``") + "`"
   }
 
-  def toPrettySQL(e: Expression): String = usePrettyExpression(e).sql
+  def toPrettySQL(e: Expression): String = e match {
+    case r: RuntimeReplaceable => r.prettySQL
```

Comment:
does this work if `RuntimeReplaceable` is not the root node?

Reply:
Ah.. no..

Reply:
https://github.com/apache/spark/pull/28420/files#diff-6f4edc80e2cc973e748705e85a6053b4R788-R807

fixed and added some nested tests, thanks, good catch!